src/Symfony/Bundle/FrameworkBundle/Command/CachePoolPruneCommand.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Cache\PruneableInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\DependencyInjection\Argument\RewindableGenerator;
+
+/**
+ * Cache pool pruner command.
+ */
+final class CachePoolPruneCommand extends Command
+{
+    /**
+     * @var RewindableGenerator
+     */
+    private $pools;
+
+    /**
+     * @param RewindableGenerator $pools
+     */
+    public function __construct(RewindableGenerator $pools)
```

Comment:
Got it. What about https://github.com/symfony/symfony/pull/23451#issuecomment-314921923 --- should we use reflection in the compiler pass or a `instanceof` check in the command to filter the pools? And, should we get rid of the `pools` argument completely or not?